[DOCS-14723] Add injector debug logs section to SSI troubleshooting page (#37875)
DataDog/documentation: content/en/tracing/trace_collection/single-step-apm/troubleshooting.md

3. The directory `/opt/datadog-packages/datadog-apm-inject` exists, with `stable` and `$version` subdirectories.
4. Language-specific directories exist (for example, `/opt/datadog/apm/library/java/` for Java).

> To enable debug logging during manual verification, set both `DD_TRACE_DEBUG` (for the SDK) and `DD_APM_INSTRUMENTATION_DEBUG` (for the injector) in your pod spec, then delete the pod so the new values take effect during injection. For how to enable injector debug output and read it, see [Injector debug logs](#injector-debug-logs).

Author: ```suggestion
To enable debug logs during manual verification:

1. Set the following in your pod spec:
 
   {{< code-block lang="yaml" disable_copy="true" collapsible="true" >}}
   env:
     - name: DD_TRACE_DEBUG    # debug logging for the SDK
       value: "true"
     - name: DD_APM_INSTRUMENTATION_DEBUG    # debug logging for the injector
       value: "true"
   {{< /code-block >}}
   
2. Delete the pod to enable debug logs during injection.

For how to enable injector debug output and read it, see [Injector debug logs](#injector-debug-logs).
```

   
2. Delete the pod to enable debug logs during injection.

> For how to enable injector debug output and read it, see [Injector debug logs](#injector-debug-logs).

Reviewer: Could just delete this since it comes immediately next?
```suggestion
```

<DEBUG> ... [linux/env_injector.c:38] injection config: DD_TAGS=_dd.injection.mode:k8s
<DEBUG> ... [linux/env_injector.c:38] injection config: DD_INJECTION_ENABLED=tracer
<DEBUG> ... [linux/env_injector.c:38] injection config: NODE_OPTIONS=--require /opt/datadog/apm/library/js/node_modules/dd-trace/init.js
<INFO>  ... [./libinject.c:81] injection duration: 1.066500 ms
<DEBUG> ... [injection_metadata.c:101] sending injection-metadata telemetry: result='0', result_reason='injection completed successfully'
<DEBUG> ... [./libinject.c:231] injector finished
```

| Log line | Meaning |
| --- | --- |
| `process_exe: '<exe>'` / `Main executable path: '<path>'` | The process being evaluated. |
| `Workload selection allowed injection: continuing` | Policy permitted injection. |
| `Succesfully loaded policy: 'requirements.bin...'` | The language requirements policy for the detected runtime was loaded. |
| `detected language: '<lang>'` / `detected language version: '<version>'` | The runtime was identified (`nodejs`, `java`, `python`, `ruby`, `dotnet`, `php`) and its version was read. |

> | `injection config: <VAR>=<value>` | Each environment variable the injector set. Seeing `DD_INJECTION_ENABLED=tracer` plus the language-specific variable (`NODE_OPTIONS`, `JAVA_TOOL_OPTIONS`, and so on) confirms the tracing SDK was loaded. |

Author: ```suggestion
| `injection config: <VAR>=<value>` | Each environment variable the injector set. Seeing `DD_INJECTION_ENABLED=tracer` plus the language-specific variable (for example, `NODE_OPTIONS` and `JAVA_TOOL_OPTIONS`) confirms the tracing SDK was loaded. |
```

> | `injection config: <VAR>=<value>` | Each environment variable the injector set. Seeing `DD_INJECTION_ENABLED=tracer` plus the language-specific variable (for example, `NODE_OPTIONS` and `JAVA_TOOL_OPTIONS`) confirms the tracing SDK was loaded. |

Reviewer: ```suggestion
| `injection config: <VAR>=<value>` | Each environment variable the injector set. Seeing `DD_INJECTION_ENABLED=tracer` plus the language-specific variable (for example, `NODE_OPTIONS` or `JAVA_TOOL_OPTIONS`) confirms the tracing SDK was loaded. |
```
